### PR TITLE
Allow writes to caches when using --parallel 

### PR DESCRIPTION
**Pull request by Kotarski**

### CODE REVIEW

File: blockwork/transforms/transform.py [hunk 1]
```diff
@@ -981,19 +986,33 @@ def _input_hash(self) -> str:
...
+            ifield = field.default
+            tf._serial_interfaces[field.name] = (
+                ifield.direction,
+                SerialInterface(spec["ifaces"][field.name]),
+            )
+        return tf
+
+    def run(self, ctx: "Context") -> TSerialTransformResult:
+        """Run the transform in a container."""
+        start = time.time()
```
Comment (Intuity) [inline blockwork/transforms/transform.py:1015]: Not using `datetime.now()`? I've found the datetime timedelta objects quite nice to work with in the past

File: blockwork/transforms/transform.py [hunk 2]
```diff
@@ -1003,15 +1022,15 @@ def run_serialized(ctx: "Context", spec: TSerialTransform):
...
+        for tool_def in self.tools:
             tool = tool_def()
             tool_instances[tool.name] = tool.default
             container.add_tool(tool)
 
         # Bind interfaces to container
         interface_values: dict[str, Any] = {}
 
-        for field in fields(cast(Any, cls)):
+        for field in fields(cast(Any, self)):
```
Comment (Intuity) [inline blockwork/transforms/transform.py:1033]: What does this `cast` magic do?

File: blockwork/transforms/transform.py [hunk 3]
```diff
@@ -1020,21 +1039,24 @@ def run_serialized(ctx: "Context", spec: TSerialTransform):
                     ", e.g. `myinput: Path = Transform.IN()`"
                 )
             ifield = field.default
-            serial = SerialInterface(spec["ifaces"][field.name])
+            serial = self._serial_interfaces[field.name][1]
```
Comment (Intuity) [inline blockwork/transforms/transform.py:1042]: Magic number when unpacking the tuple?

File: blockwork/workflows/workflow.py [hunk 4]
```diff
@@ -352,12 +351,11 @@ def _run(
...
                     logging.info("Running transform: %s", transform)
-                    start = time.time()
-                    transform.run(ctx)
-                    stop = time.time()
-                    duration = stop - start
+                    result = transform.run(ctx)
                     run_transforms.add(transform)
-                    if is_caching and Cache.store_transform_to_any(ctx, transform, duration):
+                    if is_caching and Cache.store_transform_to_any(
+                        ctx, transform, result["run_time"]
```
Comment (Intuity) [inline blockwork/workflows/workflow.py:357]: Why is this not `result.run_time`?

**Review by Intuity — APPROVED** [review]

File: blockwork/transforms/transform.py (hunk 2 above)
Comment (Kotarski) [reply blockwork/transforms/transform.py:1035]: Added a comment
>      # Cast to Any to satisfy type checkers as the base Transform class will
>      # not be a dataclass, but subclasses will.

File: blockwork/transforms/transform.py (hunk 3 above)
Comment (Kotarski) [reply blockwork/transforms/transform.py]: Updated

File: blockwork/transforms/transform.py (hunk 1 above)
Comment (Kotarski) [reply blockwork/transforms/transform.py:1015]: I want to keep the result object as simple jsonable data so just want a float here - in which case this is quite a bit simpler than using datetime

**Pull request merged**